src/gen/conditions.rs

```diff
@@ -2104,7 +2117,7 @@ fn test_create_coin_amount_exceeds_max() {
         cond_test("((({h1} ({h2} (123 (((51 ({h2} (0x010000000000000000 )))))")
             .unwrap_err()
             .1,
-        ErrorCode::AmountExceedsMaximum
+        ErrorCode::InvalidCoinAmount
```

Comment:
`ErrorCode::AmountExceedsMaximum` and `ErrorCode::NegativeAmount` map to some logic in `chia-blockchain` so it would be worth preserving them.

Reply:
I restored these error codes